### PR TITLE
[GOOWOO-373] Update Google API Queries to allow campaigns to be created/edited without an Merchant Center account 2

### PR DESCRIPTION
**Pull request by jamesmorrison**
### Changes proposed in this Pull Request:

Closes https://linear.app/a8c/issue/GOOWOO-373/update-google-api-queries-to-allow-campaigns-to-be-creatededited

Follow up to QA feedback on PR 3178: https://github.com/woocommerce/google-listings-and-ads/pull/3178#issuecomment-3689090697

### Screenshots:

N/A

## Test 1: Create Campaign without Merchant Account

1. Go to: `Marketing → Google for WooCommerce → Dashboard`
2. Click "Add campaign"
3. Fill form (name, budget, countries)
4. Click "Create campaign"
5. **Verify**: Success message appears
6. **Verify**: Campaign appears in dashboard list

## Test 2: Edit Campaign without Merchant Account

1. Go to: `Marketing → Google for WooCommerce → Dashboard`
2. Find campaign created in Test 1
3. Click "Edit"
4. Change name/budget/status
5. Click "Save"
6. **Verify**: Success message appears
7. **Verify**: Changes are reflected in dashboard

## Test 3: Budget Recommendations without Merchant Account

1. Go to: `Marketing → Google for WooCommerce → Dashboard`
2. Click "Add campaign"
3. Enter budget amount in the form
4. **Verify**: Budget recommendations appear (Low/Recommended/High options with metrics)
5. **Verify**: No error messages displayed
6. **Verify**: No errors in browser console (F12)

## Test 4: Edge Case - Edit Campaign Created with Merchant Account after Merchant Account disconnected

1. Go to: `Marketing → Google for WooCommerce → Dashboard`
2. Find campaign created in Setup (before Merchant Account disconnect)
3. Click "Edit"
4. Make any change (name, budget, or status)
5. Click "Save"
6. **Verify**: Error message appears: "Cannot edit campaign: This campaign was created with a Merchant Center account, but the account is now disconnected..."
7. **Verify**: Campaign is NOT updated (changes not saved)

### CODE REVIEW

File: tests/e2e/specs/add-paid-campaigns/add-paid-campaigns.test.js [hunk 1]
```diff
@@ -442,13 +442,40 @@ test.describe( 'Set up Ads account', () => {
...
+		} );
+
 		test( 'It should show the campaign creation success message', async () => {
 			await setupBudgetPage.fillBudget( '6' );
 			await setupBudgetPage.getCreateCampaignButton().click();
 
 			const cancelButton = page.getByRole( 'button', { name: 'Cancel' } );
 			await expect(
-				page.getByText( 'This offer won’t last long!' )
+				page.getByText( "This offer won't last long!" )
```
Comment (asvinb) [inline tests/e2e/specs/add-paid-campaigns/add-paid-campaigns.test.js]: @jamesmorrison Why that change please?

File: tests/e2e/specs/onboarding/step-2-create-campaign-ads-account-only.test.js [hunk 2]
```diff
@@ -397,7 +409,7 @@ test.describe( 'Create campaign for Ads only merchants', () => {
 					} );
 
 					await expect(
-						page.getByText( 'This offer won’t last long!' )
+						page.getByText( "This offer won't last long!" )
```
Comment (asvinb) [inline tests/e2e/specs/onboarding/step-2-create-campaign-ads-account-only.test.js]: Same as my previous comment.

**Review by jjgrainger — APPROVED** [review]

File: tests/e2e/specs/onboarding/step-2-create-campaign-ads-account-only.test.js (hunk 2 above)
Comment (jamesmorrison) [reply tests/e2e/specs/onboarding/step-2-create-campaign-ads-account-only.test.js]: Sorry, AI was helping me with tests and I missed this unexpected change; I have reverted it.

File: tests/e2e/specs/add-paid-campaigns/add-paid-campaigns.test.js (hunk 1 above)
Comment (jamesmorrison) [reply tests/e2e/specs/add-paid-campaigns/add-paid-campaigns.test.js]: Sorry, AI was helping me with tests and I missed this unexpected change; I have reverted it.
